src/gui/render/gpu_data/texture.hpp

```diff
@@ -0,0 +1,127 @@
+
+#pragma once
+
+#include "global_context.hpp"
+#include "logging.hpp"
+#include "types.hpp"
+
+#include <GL/glew.h>
+#include <glm/glm.hpp>
+
+#include <vector>
+
+namespace gui {
+
+namespace gpu_data {
+
+class Texture1D {
+ private:
+    GLuint texture_ID_;
+
+ public:
+    inline Texture1D() {
+        // create one texture and save the id to color_texture_
+        glGenTextures(1, &texture_ID_);
+    }
+
+    inline void
+    bind(GLuint texture_index) const {
+        glActiveTexture(GL_TEXTURE0 + texture_index);
+        glBindTexture(GL_TEXTURE_1D, texture_ID_);
+    }
+
+    inline Texture1D(const std::vector<ColorFloat>& color_data) {
+        // create one texture and save the id to color_texture_
+        glGenTextures(1, &texture_ID_);
+        // bind to color_texture_
+        glBindTexture(GL_TEXTURE_1D, texture_ID_);
+        glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
+        // set texture warp. values outside of texture bounds are clamped to
+        // edge of texture
+        glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+        glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+        // no minimapping so this one don't matter
+        glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
+        // set magnification filter to linear. Interpolate between pixels in mat
+        glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+        // load and generate the texture
+        glTexImage1D(
+            GL_TEXTURE_1D, 0, GL_RGBA32F, color_data.size(), 0, GL_RGBA, GL_FLOAT,
+            color_data.data()
+        );
+        glGenerateMipmap(GL_TEXTURE_1D);
+    }
+
+    inline ~Texture1D() { glDeleteTextures(1, &texture_ID_); }
+
+    [[nodiscard]] inline GLuint
+    value() const {
+        return texture_ID_;
+    }
+};
+
+struct texture2D_data {
```

Comment:
```suggestion
struct texture2D_data_t {
```

also move inside of `Texture2D`